lib/driver_sht41/include/driver/sht41.h

```diff
@@ -12,21 +12,70 @@
  *
  */
 
+// i2c address for sht41 driver
+#define SHT41_I2C_ADDR 0x44
+
+// time needed for sht41 driver to complete sampling temperature and humidity
+#define SHT41_SAMPLING_TIME_MS 15
+
+// PICO_CONFIG: SHT41_POLL_TIME_MS, time in between reading temperature and humidity, type=int, default=500, group=driver_sht41
+#ifndef SHT41_POLL_TIME_MS
+#define SHT41_POLL_TIME_MS 500
+#endif
+
+// maximum duration of time for a valid sht41 temperature and humidity data before it's considered invalid
+#define SHT41_INVALID_TIMEOUT_MS (SHT41_POLL_TIME_MS + 100)
+
+/**
+ * @brief types of error sht41 driver could have
+ *
+ */
 typedef enum {
     SHT41_ERROR_TIMER_SCHEDULE_FULL = 1,
     SHT41_ERROR_CRC_INVALID,
     SHT41_ERROR_I2C_COMPLAINT
 } sht41_error_code;
 
+/**
+ * @brief boolean if temperature and humidity data is ready to read
+ *
+ */
 extern volatile bool sht41_temp_rh_set_on_read;
 
-typedef void (*sht41_error_cb)(const sht41_error_code error_type, uint32_t i2c_error_code);
+/**
+ * @brief error callback.
+ *
+ */
+typedef void (*sht41_error_cb)(const sht41_error_code error_type);
 
-void sht41_init(sht41_error_cb board_error_cb);
+/**
+ * @brief Initializes sht41 driver
+ *
+ * @param board_error_cb callback on error
+ * @param board the i2c bus number for sht41 driver
```

Comment:
Board isn't a great name for this parameter, since that's the name of a specific i2c bus, which might not always be the case.
bus_num might be better